src/uct/ib/ud/base/ud_ep.c

```diff
@@ -665,23 +679,37 @@ static uct_ud_ep_t *uct_ud_ep_create_passive(uct_ud_iface_t *iface, uct_ud_ctl_h
     /* create new endpoint */
     params.field_mask = UCT_EP_PARAM_FIELD_IFACE;
     params.iface      = &iface->super.super.super;
-    status = uct_ep_create(&params, &ep_h);
-    ucs_assert_always(status == UCS_OK);
+    status            = uct_ep_create(&params, &ep_h);
+    if (status != UCS_OK) {
+        return NULL;
+    }
+
     ep = ucs_derived_of(ep_h, uct_ud_ep_t);
 
     status = uct_ep_connect_to_ep(ep_h, (void*)uct_ud_creq_ib_addr(ctl),
                                   (void*)&ctl->conn_req.ep_addr);
-    ucs_assert_always(status == UCS_OK);
+    if (status != UCS_OK) {
+        goto err_unbind_from_iface;
+    }
 
     ep->path_index = ctl->conn_req.path_index;
 
     uct_ud_ep_set_state(ep, UCT_UD_EP_FLAG_PRIVATE);
 
     ep->conn_sn = ctl->conn_req.conn_sn;
-    uct_ud_iface_cep_insert_ep(iface, uct_ud_creq_ib_addr(ctl),
-                               &ctl->conn_req.ep_addr.iface_addr,
-                               ep->path_index, ctl->conn_req.conn_sn, ep);
+    status      = uct_ud_iface_cep_insert_ep(iface, uct_ud_creq_ib_addr(ctl),
+                                             &ctl->conn_req.ep_addr.iface_addr,
+                                             ep->path_index,
+                                             ctl->conn_req.conn_sn, ep);
+    if (status != UCS_OK) {
+        goto err_unbind_from_iface;
+    }
+
     return ep;
+
+err_unbind_from_iface:
+    uct_ud_ep_unbind_from_iface(&ep->super.super);
```

Comment:
don't we need to destroy the endpoint here? since it was created in line 682
then can restore the name `uct_ud_ep_disconnect_from_iface`

Reply:
> don't we need to destroy the endpoint here? since it was created in line 682

it is destroyed in `uct_ud_ep_unbind_from_iface()`

Reply:
i see uct_ud_ep_unbind_from_iface only reset, where does it destroy?

Reply:
yes, you're right, it was a leek. good catch.
need to add destroying EP